src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNIndexShard.java

```diff
@@ -0,0 +1,124 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.knn.index;
+
+import com.amazon.opendistroforelasticsearch.knn.index.codec.KNNCodecUtil;
+import com.amazon.opendistroforelasticsearch.knn.index.v1736.KNNIndex;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.index.FilterLeafReader;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.SegmentInfo;
+import org.apache.lucene.index.SegmentReader;
+import org.elasticsearch.index.engine.Engine;
+import org.elasticsearch.index.shard.IndexShard;
+import org.elasticsearch.index.shard.ShardPath;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * KNNIndexShard wraps IndexShard and adds methods to perform k-NN related operations against the shard
+ */
+public class KNNIndexShard {
+    private IndexShard indexShard;
+    private KNNIndexCache knnIndexCache;
+
+    private static Logger logger = LogManager.getLogger(KNNIndexShard.class);
+
+    /**
+     * Constructor to generate KNNIndexShard. We do not perform validation that the Index the IndexShard is from
```

Comment:
remove "the Index" and just leave "the IndexShard"?

Reply:
I think I wiil change to "We do not perform validation that the index the shard is from"